bcipy/gui/BCInterface.py

```diff
@@ -327,6 +328,54 @@ def check_input(self) -> bool:
             return False
         return True
 
+    def user_id_validator(self) -> bool:
+        """Check User ID Requirements. User ID must follow the following Requirements:
+
+        1. Maximum length of 25 letters or numbers
+        2. Minimum length of at least one letter or  number
+        3. No special characters
+        4. No spaces
+        """
+
+        # Update based on current inputs
+        self.user = self.user_input.currentText()
+        allowed_characters = re.compile('[^0-9a-zA-Z]+')
+
+        # Check the length of minimum one character up to 25 characters
+        try:
+            if len(self.user) == 0 or len(self.user) > 25:
+                self.throw_alert_message(
+                    title='BciPy Alert',
+                    message='User ID must be of at least one or up to 25 non-special characters',
+                    message_type=AlertMessageType.INFO,
+                    okay_to_exit=True)
+                return False
+        # Check for white space
+            if re.match(r"^(?=.*[\s])", self.user):
+                self.throw_alert_message(
+                    title='BciPy Alert',
+                    message='User ID cannot contain white spaces',
+                    message_type=AlertMessageType.INFO,
+                    okay_to_exit=True)
+                return False
+        # Check for special characters
+            if allowed_characters.search(self.user):
```

Comment:
Here it looks like you are failing validation if there are _any_ allowed characters. Shouldn't this be the opposite and only raise the alert if _all_ of the characters are allowed?

Reply:
Sorry, my bad.